test/v2/core/wsgETH.spec.js

```diff
@@ -0,0 +1,198 @@
+      },
+      approveData,
+    );
+
+    const {r, s, v} = ethers.utils.splitSignature(signature);
+
+    await expect(wsgEth.connect(alice).depositWithSignature(parseEther("1"), alice.address, deadline, false, v, r, s))
+      .to.be.emit(wsgEth, "Transfer")
+      .withArgs(ethers.constants.AddressZero, alice.address, parseEther("1"));
+  });
+
+  it("price per share", async () => {
+    const splitterAddresses = [deployer.address, multiSig.address, wsgEth.address];
+    const splitterValues = [6, 3, 31];
+
+    const PaymentSplitter = await ethers.getContractFactory("PaymentSplitter");
+    const paymentSplitter = await PaymentSplitter.deploy(splitterAddresses, splitterValues);
+    await paymentSplitter.deployed();
+
+    const rolloverVirtual = "1080000000000000000";
+    const vETH2Addr = "0x898bad2774eb97cf6b94605677f43b41871410b1";
+
+    const Withdrawals = await ethers.getContractFactory("Withdrawals");
+    const withdrawals = await Withdrawals.deploy(vETH2Addr, rolloverVirtual);
+    await withdrawals.deployed();
+
+    const numValidators = 1000;
+    const adminFee = 0;
+
+    const addresses = [
+      ethers.constants.AddressZero, // fee splitter
+      sgEth.address, // sgETH address
+      wsgEth.address, // wsgETH address
+      multiSig.address, // government address
+      ethers.constants.AddressZero, // deposit contract address - can't find deposit contract - using dummy address
+    ];
+
+    // add secondary minter contract / eoa
+    const Minter = await ethers.getContractFactory("SharedDepositMinterV2");
+    const minter = await Minter.deploy(numValidators, adminFee, addresses);
+    await minter.deployed();
+
+    const RewardsReceiver = await ethers.getContractFactory("RewardsReceiver");
+    const rewardsReceiver = await RewardsReceiver.deploy(withdrawals.address, [
+      sgEth.address,
+      wsgEth.address,
+      paymentSplitter.address,
+      minter.address,
+    ]);
+    await rewardsReceiver.deployed();
+
+    await sgEth.addMinter(minter.address);
+
+    // approve sgEth to alice
+    await sgEth.approve(wsgEth.address, parseEther("2"));
+    await wsgEth.deposit(parseEther("2"), alice.address);
+
+    console.log(formatEther(await wsgEth.pricePerShare()));
+
+    await expect(wsgEth.connect(alice).redeem(parseEther("0.5"), alice.address, alice.address))
+      .to.be.emit(wsgEth, "Withdraw")
+      .withArgs(alice.address, alice.address, alice.address, parseEther("0.5"), parseEther("0.5"));
+
+    // deposit to rewardReceiver to simulate reward
+    await deployer.sendTransaction({
+      to: rewardsReceiver.address,
+      value: parseEther("1"),
+    });
+    // sends 60% of sgEth to WSGEth contract - so current rate is 1.5/2.1
+    console.log(await rewardsReceiver.state());
+    await rewardsReceiver.work();
+
+    await expect(wsgEth.syncRewards()).to.be.revertedWith("SyncError()");
+    // increase time by reward cycle
+    await time.increase(24 * 60 * 60);
+    await wsgEth.syncRewards();
```

Comment:
is this call necessary? it should be triggered when redeem/withdraw is called automatically. and its a error condition if it needs to be manually triggered instead.

Reply:
actually syncRewards calls when user call another functions such as deposit and withdraw/redeem.
so actually not sure this function is called on time. But I should call this function to test increment of reward

Reply:
it should work without the direct call. when the user calls another function 
or try to call  the following again after sync rewards: 
```
    // deposit to rewardReceiver to simulate reward
    await deployer.sendTransaction({
      to: rewardsReceiver.address,
      value: parseEther("1"),
    });
    // sends 60% of sgEth to WSGEth contract - so current rate is 1.5/2.1
    await rewardsReceiver.work();
```